java-frontend/src/main/java/org/sonar/java/se/checks/SECheck.java

```diff
@@ -75,6 +80,36 @@ public void reportIssue(Tree tree, String message, Set<List<JavaFileScannerConte
       .orElse(new SEIssue(tree, message, secondary)));
   }
 
+  public static List<JavaFileScannerContext.Location> flow(ExplodedGraph.Node currentNode, SymbolicValue currentVal ) {
+    List<JavaFileScannerContext.Location> flow = new ArrayList<>();
+    ExplodedGraph.Node node = currentNode;
+    Symbol lastEvaluated = currentNode.programState.getLastEvaluated();
+    while (node != null) {
+      ExplodedGraph.Node finalNode = node;
+      if(finalNode.programPoint.syntaxTree() != null) {
+        node.getLearnedConstraints().stream()
+          .map(lc->lc.sv)
+          .filter(sv -> sv.equals(currentVal))
+          .findFirst()
```

Comment:
main difference between `findAny()` and `findFirst()` is that second one is imposing more constraints, which has performance implications for parallel streams, but also semantically it makes me thing why we want exactly first and not any element.

Btw, if there should be only one, shouldn't that be asserted somehow? (i.e. `Preconditions.checkState` or similar)